File: tests/xcalablemp/NOT-IMPLEMENTED/gmove_in_out/F/out2.f90
! testp018.f
! gmove指示文のテスト

      program main
      include 'xmp_lib.h'
      integer,parameter:: N=1000
!$xmp nodes p(*)
!$xmp template t1(N,N)
!$xmp template t2(N,N)
!$xmp distribute t1(*,block) onto p
!$xmp distribute t2(cyclic,*) onto p
      integer a1(N), a2(N), sa
      real*8  b1(N), b2(N), sb
      real*4  c1(N), c2(N), sc
!$xmp align a1(i) with t1(*,i)
!$xmp align b1(i) with t1(*,i)
!$xmp align c1(i) with t1(*,i)
!$xmp align a2(i) with t2(i,*)
!$xmp align b2(i) with t2(i,*)
!$xmp align c2(i) with t2(i,*)
      integer procs
      character(len=2) result

!$xmp loop on t1(:,i)
      do i=1, N
         a1(i) = i
         b1(i) = dble(i)
         c1(i) = real(i)
      enddo

!$xmp looop on t2(i,:)
      do i=1, N
         a2(i) = 1
         b2(i) = 1.0
         c2(i) = 1.0
      enddo
      
      procs = xmp_num_nodes()
      do j=1, procs
         if(j.eq.xmp_node_num()) then
!$xmp gmove out
            a1(:) = a2(:)
!$xmp gmove out
            c1(:) = c2(:)
!$xmp gmove out
            b1(:) = b2(:)
         endif
      enddo

      sa = 0
      sb = 0.0
      sc = 0.0
!$xmp loop on t1(:,i) reduction(+:sa,sb,sc)
      do i=1, N
         sa = sa+a1(i)
         sb = sb+b1(i)
         sc = sc+c1(i)
      enddo

      result = 'OK'
      if(sa .ne. 1000 .or.
     $     abs(sb-1000.0) .gt. 0.000000001 .or.
     $     abs(sc-1000.0) .gt. 0.0001) then
         result = 'NG'
      endif

      print *, xmp_node_num(), 'testp018.f ', result
      
      end

File: tests/xcalablemp/NOT-IMPLEMENTED/39/out2.c
/*testp018.c*/
/*gmove指示文のテスト*/
#include<mpi.h>
#include<xmp.h>
#include<stdio.h>
#include<stdlib.h>   
static const int N=1000;
#pragma xmp nodes p(*)
#pragma xmp template t1(0:N-1,0:N-1)
#pragma xmp template t2(0:N-1,0:N-1)
#pragma xmp distribute t1(*,block) onto p
#pragma xmp distribute t2(cyclic,*) onto p
int a1[N],a2[N],sa;
double b1[N],b2[N],sb;
float c1[N],c2[N],sc;
#pragma xmp align a1[i] with t1(*,i)
#pragma xmp align b1[i] with t1(*,i)
#pragma xmp align c1[i] with t1(*,i)
#pragma xmp align a2[i] with t2(i,*)
#pragma xmp align b2[i] with t2(i,*)
#pragma xmp align c2[i] with t2(i,*)
int procs;
int i,j;
char *result;
int main(void){

#pragma xmp loop on t1(:,i)
   for(i=0;i<N;i++){
      a1[i] = i;
      b1[i] = (double)i;
      c1[i] = (float)i;
   }

#pragma xmp loop on t2(i,:)
   for(i=0;i<N;i++){
      a2[i] = 1;
      b2[i] = 1.0;
      c2[i] = 1.0;
   }

   procs = xmp_num_nodes();
   for(j=1;j<procs+1;j++){
      if(j == xmp_node_num()){
#pragma xmp gmove out 
         a1[:] = a2[:];
#pragma xmp gmove out 
         c1[:] = c2[:];
#pragma xmp gmove out 
         b1[:] = b2[:];
      }
   }

   sa = 0;
   sb = 0.0;
   sc = 0.0;
#pragma xmp loop on t1(:,i) reduction(+:sa,sb,sc)
   for(i=0;i<N;i++){
      sa = sa+a1[i];
      sb = sb+b1[i];
      sc = sc+c1[i];
   }

   result = "OK";
   if(sa != 1000||abs(sb-1000.0) > 0.000000001|| abs(sc-1000.0) > 0.0001){
      result = "NG";
   }

   printf("%d %s %s\n",xmp_node_num(),"testp018.c",result); 
   return 0;
}
      
         
      
   
















